frontend/src/pages/NewRunV2.tsx

```diff
@@ -81,89 +86,25 @@ function NewRunV2(props: NewRunV2Props) {
   const [errorMessage, setErrorMessage] = useState('');
   const [isParameterValid, setIsParameterValid] = useState(false);
   const [clonedRuntimeConfig, setClonedRuntimeConfig] = useState<PipelineSpecRuntimeConfig>({});
-  const [usePipelineFromRunLabel, setUsePipelineFromRunLabel] = useState(
-    'Using pipeline from cloned run',
-  );
 
   // TODO(zijianjoy): If creating run from Experiment Page or RunList Page, there is no pipelineId/Version.
   const urlParser = new URLParser(props);
-  const originalRunId = urlParser.get(QUERY_PARAMS.cloneFromRun);
-  const pipelineId = urlParser.get(QUERY_PARAMS.pipelineId);
-  const pipelineVersionIdParam = urlParser.get(QUERY_PARAMS.pipelineVersionId);
-  const pipelineDetailsUrl = originalRunId
+  const usePipelineFromRunLabel = 'Using pipeline from existing run.';
+  const { apiRun, apiPipeline, apiPipelineVersion } = props;
+  const pipelineDetailsUrl = props.originalRunId
     ? RoutePage.PIPELINE_DETAILS.replace(
         ':' + RouteParams.pipelineId + '/version/:' + RouteParams.pipelineVersionId + '?',
         '',
-      ) + urlParser.build({ [QUERY_PARAMS.fromRunId]: originalRunId })
+      ) + urlParser.build({ [QUERY_PARAMS.fromRunId]: props.originalRunId })
     : '';
 
-  // Retrieve Run Detail using RunID from backend
-  const { data: apiRun } = useQuery<ApiRunDetail, Error>(
-    ['ApiRun', originalRunId],
-    () => {
-      if (!originalRunId) {
-        throw new Error('No Run ID');
-      }
-      const originalRun = Apis.runServiceApi.getRun(originalRunId);
-      return originalRun;
-    },
-    { enabled: !!originalRunId, staleTime: Infinity },
-  );
+  const templateString = props.templateString;
+  const isTemplatePullSuccess = templateString !== '';
```

Comment:
What if `templateString` is undefined? The value of `isTemplatePullSuccess` is true in this case.

Reply:
change to const isTemplatePullSuccess = templateString !== '' || undefined;

Reply:
This doesn't seem to have an effect to situation when `templateString` is `undefined`. Suggest the following:

```
isTemplatePullSuccess = templateString && templateString !== '';
```